src/de/mod.rs

```diff
@@ -1085,7 +1271,7 @@ mod tests {
         assert_eq!(
             crate::from_str::<Xy>(r#"[10]"#),
             Err(crate::de::Error::CustomErrorWithMessage(
-                heapless::String::from_str(
+                core::str::FromStr::from_str(
```

Comment:
Can't we just call `"invalid length 1, ....".parse().unwrap()? I think all of these prefixes are extraneous.